src/components/FeatureThresholdPanel.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { ReactElement, ReactNode, useEffect } from "react";
+import { FeatureThreshold } from "../colorizer/ColorizeCanvas";
+import { Dataset } from "../colorizer";
+import { Card, List, Select } from "antd";
+import styled from "styled-components";
+import LabeledRangeSlider from "./LabeledRangeSlider";
+import { CloseOutlined, FilterOutlined } from "@ant-design/icons";
+import IconButton from "./IconButton";
+
+const PanelContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  gap: 6px;
+  height: 100%;
+`;
+
+const FiltersCard = styled(Card)`
+  overflow-y: auto;
+  height: 100%;
+
+  & .ant-card-body {
+    padding-top: 0;
+    padding-bottom: 0;
+    height: 100%;
+  }
+`;
+
+const EmptyListTextContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+`;
+
+type FeatureThresholdPanelProps = {
+  featureThresholds: FeatureThreshold[];
+  onChange: (thresholds: FeatureThreshold[]) => void;
+  dataset: Dataset | null;
+  disabled?: boolean;
+};
+const defaultProps: Partial<FeatureThresholdPanelProps> = {
+  disabled: false,
+};
+
+/**
+ * A React component for adding, removing, and editing thresholds on features in a dataset.
+ */
+export default function FeatureThresholdPanel(inputProps: FeatureThresholdPanelProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<FeatureThresholdPanelProps>;
+
+  // Clear thresholds for features that don't exist when the dataset changes.
+  // TODO: Show these thresholds as disabled, rather than removing them.
+  useEffect(() => {
+    const newThresholds = props.featureThresholds.filter((t) => props.dataset?.featureNames.includes(t.featureName));
+    props.onChange(newThresholds);
+  }, [props.dataset]);
+
+  /** Handle the user selecting new features. */
+  const onSelectionsChanged = (selections: string[]): void => {
+    const newThresholds: FeatureThreshold[] = [];
+    selections.forEach((featureName) => {
+      // Set up default values for any new selected features, otherwise keep old thresholds
+      const existingThreshold = props.featureThresholds.find((t) => t.featureName === featureName);
+      if (existingThreshold) {
+        newThresholds.push(existingThreshold);
+      } else {
+        const featureData = props.dataset?.features[featureName];
+        if (featureData) {
+          newThresholds.push({ featureName, min: featureData.min, max: featureData.max });
+        }
+      }
+    });
```

Comment:
Suggestion: this `forEach` could become a `map` from `string[]` to `FeatureThreshold[]`.